Add support for database engine SAP Hana

axuew opened the pull request:
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

Support hana Time Grain
![image](https://user-images.githubusercontent.com/44701753/67068254-0c078880-f1ac-11e9-9a4a-d3d2b51e74a1.png)

This feature requires the addition of a third library: sqlalchemy-hana and hdbcli

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

Commit:
Add hana

mistercrunch commented on superset/jinja_context.py:
```diff
@@ -288,3 +289,15 @@ class HiveTemplateProcessor(PrestoTemplateProcessor):
     TP = template_processors.get(database.backend, BaseTemplateProcessor)
     return TP(database=database, table=table, query=query, **kwargs)
+
+def get_data_permission(sql_statement: str) -> List[str]:
```
Does this have anything to do with Hana?

mistercrunch commented on docs/installation.rst:
```diff
@@ -402,6 +402,9 @@ Here's a list of some of the recommended packages.
 | Vertica          | ``pip install                         |  ``vertica+vertica_python://``                  |
 |                  | sqlalchemy-vertica-python``           |                                                 |
 +------------------+---------------------------------------+-------------------------------------------------+
+| Hana             | ``pip install hdbcli``                |  ``hana://``                  |
```
based on other suggested changes using setuptools' `extras_require`, this should be `pip install superset[hana]`

mistercrunch reviewed (changes requested):
Second take on https://github.com/apache/incubator-superset/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+hana ? Can you close the original? 

By creating a new PR and not referencing the previous one, we loose all the context that was builtup previously. Next time please just append to the existing PR.

Also please take the time to write a good PR title that summarizes your PR next time.

axuew committed:
Restore previous version

Restore previous version

Delete the following two items：
hdbcli==2.4.162
sqlalchemy-hana==0.4.0

axuew committed:
Delete jinja_context.py

Remove non-hana functional code
